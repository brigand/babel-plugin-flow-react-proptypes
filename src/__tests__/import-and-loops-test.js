const babel = require('babel-core');
const content = `
// @flow

import React from 'react';
import type { ExternalType } from '../types';

export type T = {
    flag: boolean,
    list: Array<ExternalType>,
};

const C = ({
    flag,
    list,
}: T) => {
    return flag
        ? <div>yes</div>
        : <Select>
            {list.map((e: ExternalType) => {
                return (
                    <option value={e.id} key={e.id}>
                    </option>
                );
            })}
        </Select>;
};

export default C;
`;

it('import-and-loops', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
