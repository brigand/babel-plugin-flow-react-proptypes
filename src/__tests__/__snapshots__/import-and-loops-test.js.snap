exports[`test import-and-loops 1`] = `
"\'use strict\';

Object.defineProperty(exports, \"__esModule\", {
    value: true
});

var _react = require(\'react\');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_ExternalType = require(\'../types\').babelPluginFlowReactPropTypes_proptype_ExternalType || require(\'prop-types\').any;

if (typeof exports !== \'undefined\') Object.defineProperty(exports, \'babelPluginFlowReactPropTypes_proptype_T\', {
    value: {
        flag: require(\'prop-types\').bool.isRequired,
        list: require(\'prop-types\').arrayOf(typeof babelPluginFlowReactPropTypes_proptype_ExternalType === \'function\' ? babelPluginFlowReactPropTypes_proptype_ExternalType : require(\'prop-types\').shape(babelPluginFlowReactPropTypes_proptype_ExternalType)).isRequired
    }
});


var C = function C(_ref) {
    var flag = _ref.flag,
        list = _ref.list;

    return flag ? _react2.default.createElement(
        \'div\',
        null,
        \'yes\'
    ) : _react2.default.createElement(
        Select,
        null,
        list.map(function (e) {
            return _react2.default.createElement(\'option\', { value: e.id, key: e.id });
        })
    );
};

C.propTypes = {
    flag: require(\'prop-types\').bool.isRequired,
    list: require(\'prop-types\').arrayOf(typeof babelPluginFlowReactPropTypes_proptype_ExternalType === \'function\' ? babelPluginFlowReactPropTypes_proptype_ExternalType : require(\'prop-types\').shape(babelPluginFlowReactPropTypes_proptype_ExternalType)).isRequired
};
exports.default = C;"
`;
